From 2598c0ceb11d39e2538c9d3c6114797ea2d11148 Mon Sep 17 00:00:00 2001
From: Xu Qiang <xuqiang36@huawei.com>
Date: Sat, 11 Feb 2023 10:23:26 +0800
Subject: [PATCH 007/500] [Huawei] mm/sharepool: split meminfo_update_k2u into
 meminfo_inc_k2u and meminfo_dec_k2u.

Offering: HULK
hulk inclusion
category: other
bugzilla: https://gitee.com/openeuler/kernel/issues/I6ET9W

----------------------------------------------

Signed-off-by: Xu Qiang <xuqiang36@huawei.com>
---
 mm/share_pool.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 7a9a17feef6f..0c8d2fa603ca 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -299,16 +299,16 @@ static void meminfo_dec_usage(unsigned long size, bool huge, struct sp_meminfo *
 		atomic64_sub(size, &meminfo->alloc_nsize);
 }
 
-static void meminfo_update_k2u(unsigned long size, bool inc,
-	struct sp_meminfo *meminfo)
+static void meminfo_inc_k2u(unsigned long size, struct sp_meminfo *meminfo)
 {
-	if (inc) {
-		atomic64_add(size, &meminfo->size);
-		atomic64_add(size, &meminfo->k2u_size);
-	} else {
-		atomic64_sub(size, &meminfo->size);
-		atomic64_sub(size, &meminfo->k2u_size);
-	}
+	atomic64_add(size, &meminfo->size);
+	atomic64_add(size, &meminfo->k2u_size);
+}
+
+static void meminfo_dec_k2u(unsigned long size, struct sp_meminfo *meminfo)
+{
+	atomic64_sub(size, &meminfo->size);
+	atomic64_sub(size, &meminfo->k2u_size);
 }
 
 /* The caller should hold mmap_sem to protect master (TBD) */
@@ -782,12 +782,12 @@ static void spa_inc_usage(struct sp_area *spa)
 	case SPA_TYPE_K2TASK:
 		spa_stat.k2u_task_num += 1;
 		spa_stat.k2u_task_size += size;
-		meminfo_update_k2u(size, true, &spa->spg->meminfo);
+		meminfo_inc_k2u(size, &spa->spg->meminfo);
 		break;
 	case SPA_TYPE_K2SPG:
 		spa_stat.k2u_spg_num += 1;
 		spa_stat.k2u_spg_size += size;
-		meminfo_update_k2u(size, true, &spa->spg->meminfo);
+		meminfo_inc_k2u(size, &spa->spg->meminfo);
 		break;
 	default:
 		WARN(1, "invalid spa type");
@@ -829,12 +829,12 @@ static void spa_dec_usage(struct sp_area *spa)
 	case SPA_TYPE_K2TASK:
 		spa_stat.k2u_task_num -= 1;
 		spa_stat.k2u_task_size -= size;
-		meminfo_update_k2u(size, false, &spa->spg->meminfo);
+		meminfo_dec_k2u(size, &spa->spg->meminfo);
 		break;
 	case SPA_TYPE_K2SPG:
 		spa_stat.k2u_spg_num -= 1;
 		spa_stat.k2u_spg_size -= size;
-		meminfo_update_k2u(size, false, &spa->spg->meminfo);
+		meminfo_dec_k2u(size, &spa->spg->meminfo);
 		break;
 	default:
 		WARN(1, "invalid spa type");
-- 
2.17.1

